lib/saml/de_serializer.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+module SAML
+  class DeSerializer
+
+    ## saml_response = OneLogin::RubySaml::Response
+    def initialize(saml_response)
+      @saml_response = saml_response
+    end
+
+    def user_attributes
```

Comment:
You could also consider calling it GoldenRecord or whatever, but I do think it makes sense to have the name of the class inform what the object is rather than to act as a service.